lib/msf/core/modules/external/go/src/metasploit/module/core.go

```diff
@@ -128,10 +131,14 @@ type (
 )
 
 func LogInfo(message string) {
+	logMutex.Lock()
```

Comment:
This still doesn't look correct. The lock needs to protect 'rpcSend', not just two Log methods.

It's not about locking for logs, its about locking for access to stdout which is how messages are sent to msfconsole. If you don't, 2 messages can overlap each other causing corruption in the output steam. Something like this:

```
--- a/lib/msf/core/modules/external/go/src/metasploit/module/core.go
+++ b/lib/msf/core/modules/external/go/src/metasploit/module/core.go
@@ -98,7 +98,12 @@ func parseParams(passedParams interface{}) (map[string]interface{}, error) {
        return v, nil
 }
 
+var rpcMutex = &sync.Mutex{}
+
 func rpcSend(res interface{}) error {
+       rpcMutex.Lock()
+       defer rpcMutex.Unlock()
+
        resStr, err := json.Marshal(res)
        if err != nil {
                return err
```